### PR TITLE
Add i18n 2

### PR DESCRIPTION
**Pull request by uri-ustrell**
### don't merge this

I did a try, but only avoids using `sprintf(  )`

1. this creates the need of passing $args even if there aren't actual variables.
2. makes code harder to understand while reading just the function call `pass()` or `fail()`.
- so I agree that it's pretty much a mess

look at JSON class and database `fail()` and `pass()` calls

### CODE REVIEW

File: includes/class-wp-security-bp-database.php [hunk 1]
```diff
@@ -133,12 +133,18 @@ public function check_name() {
...
-			$message = 'Your database name ( ' . $this->db_name . ' ) is not secure enough';
-			$trigger = 'fix DB name';
-			$this->response->fail( $short_desc, $message, $trigger );
+			/* translators: %s: Name of the database */
+			$message = __( 'Your database name (%s) is not secure enough.', 'wp-security-bp' );
+			$args    = array( $this->db_name );
+			$action  = 'database-fix-name';
+
+			$this->response->fail( $message, $args, $short_desc, $action );
 		}
```
Comment (uri-ustrell) [inline includes/class-wp-security-bp-database.php:148]: here

File: includes/class-wp-security-bp-database.php [hunk 2]
```diff
@@ -133,12 +133,18 @@ public function check_name() {
...
 
 		if ( $check ) {
-			$message = 'Your database name ( ' . $this->db_name . ' ) is fine';
-			$this->response->pass( $short_desc, $message );
+			/* translators: %s: Name of the database */
+			$message = __( 'Your database name (%s) is fine.', 'wp-security-bp' );
+			$args    = array( $this->db_name );
+
+			$this->response->pass( $message, $args, $short_desc );
 		} else {
```
Comment (uri-ustrell) [inline includes/class-wp-security-bp-database.php:141]: here

File: includes/class-wp-security-bp-json.php [hunk 3]
```diff
@@ -22,75 +21,54 @@
...
-
-			/* translators: %s: message, plugin name */
-			__(
-				'%s',
-				'%s'
-			),
-			$message,
-			$this->plugin_name
-		);
+		$this->json['message']    = vsprintf( $message, $args );
```
Comment (uri-ustrell) [inline includes/class-wp-security-bp-json.php:72]: and here....
not good enough

Comment (jbaldrich) [reply includes/class-wp-security-bp-json.php:72]: 🤔 maybe this **does** the trick.

Comment (uri-ustrell) [reply includes/class-wp-security-bp-json.php:72]: it does... but lets be fair... it's a shit

**Pull request closed without merging**